internal/controller/postgrescluster/pgbackrest.go

```diff
@@ -2758,12 +2813,52 @@ func (r *Reconciler) reconcileRepos(ctx context.Context,
 	}
 
 	postgresCluster.Status.PGBackRest.Repos =
-		getRepoVolumeStatus(postgresCluster.Status.PGBackRest.Repos, repoVols, extConfigHashes,
-			replicaCreateRepo.Name)
+		getRepoVolumeStatus(postgresCluster.Status.PGBackRest.Repos, repoVols,
+			extConfigHashes, replicaCreateRepo.Name)
 
 	return replicaCreateRepo, utilerrors.NewAggregate(errors)
 }
 
+// +kubebuilder:rbac:groups="",resources="pods",verbs={list}
+
+// getRepoHostVolumeRequests gets the pgBackRest repo host volume request annotations
+// from the repo host Pod and stores them in the Postgres Cluster object's status.
+// If there is no repo host or there are no annotations, this function returns
+// without updating the status.
+func (r *Reconciler) getRepoHostVolumeRequests(ctx context.Context,
+	cluster *v1beta1.PostgresCluster) error {
+
+	pods := &corev1.PodList{}
+	err := errors.WithStack(
+		r.Client.List(ctx, pods,
+			client.InNamespace(cluster.Namespace),
+			client.MatchingLabelsSelector{
+				Selector: naming.PGBackRestDedicatedLabels(cluster.Name).AsSelector()},
+		))
+
+	if len(pods.Items) == 1 {
```

Comment:
should we have an else case here and create an error? it looks like "if len == 1" we will just return nil
```suggestion
	if err == nil && len(pods.Items) == 1 {
	...
	} else {
	    err = errors.New("blah")
	}
	return err
```